test/core.jl

```diff
@@ -5505,6 +5505,33 @@ x.u = initvalue2(Base.uniontypes(U)[1])
 x.u = initvalue(Base.uniontypes(U)[2])
 @test x.u === initvalue(Base.uniontypes(U)[2])
 
+v = Vector{Float64}(8000000);
+mutable struct UnionField2
+    x::Union{Void, Int}
+    UnionField2() = new()
+end
+@test UnionField2().x === nothing
+
+struct UnionField3
+    x::Union{Void, Int}
+    UnionField3() = new()
+end
+@test UnionField3().x === nothing
+
+mutable struct UnionField4
+    x::Union{Void, Float64}
+    y::Union{Void, Int8}
+    UnionField4() = new()
+end
+@test UnionField4().x === nothing
+
+struct UnionField5
+    x::Union{Void, Float64}
+    y::Union{Void, Int8}
+    UnionField5() = new()
+end
+@test UnionField5().x === nothing
```

Comment:
@vtjnash, this is the only case that still fails at the REPL (I can't trigger a failure when run in a script). Are there any other places that we need to update? Something separate for struct/mutable struct in datetype.c?

Reply:
Looks like an LLVM bug (https://bugs.llvm.org/show_bug.cgi?id=34940)

Reply:
Ah......so what should we do here? Merge and open an issue to track upstream? Is there some kind of patch we could carry locally? (I say that realizing it's way over my head though in terms of what a fix here would entail)

Reply:
Bump; thoughts on what to do here?

Reply:
If these tests are working (which they seem to be), we can go ahead and merge this. The bug seems to be pretty serious, and is largely unrelated to the union optimization work. But we need upstream to fix that.

Reply:
Sounds good. I'll open an issue so we can track the upstream fix (so we don't forget)